scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/signal/SignalApiImpl.java

```diff
@@ -134,10 +150,22 @@ public boolean sendSignal(String signalName) throws SignalApiException {
     public boolean sendManySignals(List<String> signalsSubList) throws SignalApiException {
         try {
             init();
+            List<String> signals = getJobSignals();
             StringBuilder actions = new StringBuilder();
-            for (String signal : signalsSubList) {
+
+            // Filter signals to add only those signals that are not already sent
+            for (String signal : signalsSubList.stream()
+                                               .filter(signal -> !signals.contains(signal))
+                                               .collect(Collectors.toList())) {
```

Comment:
again, if you use HashSet instead of ArrayList, the code will be much simpler

 actions.append("x.remove('" + READY_PREFIX + signal + "');x.add('" + signal + "');");